# HG changeset patch
# User pm
# Date 1553526380 -3600
#      Mon Mar 25 16:06:20 2019 +0100
# Node ID f57ae0772d2c32241d137787590013584173cd57
# Parent  fd0f5ca42340b1f7a36c55814d253b715da610cf
Ticket mantis 0005130: Passage en mode hors ligne lors du lancement si le serveur imap n'est pas joignable

diff --git a/comm-esr60/comm/mailnews/extensions/offline-startup/js/offlineStartup.js b/comm-esr60/comm/mailnews/extensions/offline-startup/js/offlineStartup.js
--- a/comm-esr60/comm/mailnews/extensions/offline-startup/js/offlineStartup.js
+++ b/comm-esr60/comm/mailnews/extensions/offline-startup/js/offlineStartup.js
@@ -5,6 +5,8 @@
 
 ChromeUtils.import("resource://gre/modules/Services.jsm");
 ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
+Components.utils.import("resource://gre/modules/AppConstants.jsm");
+Components.utils.import("resource:///modules/cm2TestImapM2.jsm");
 
 var kDebug              = false;
 var kOfflineStartupPref = "offline.startup_state";
@@ -108,6 +110,23 @@
       if (result != 1 && wasOffline)
         Services.prefs.setBoolPref("mailnews.playback_offline", true);
     }
+    
+    if (1==this.cm2imap ||
+        "linux"==AppConstants.platform){
+      this.testConnexionImap();
+      this.cm2imap=1;
+    }
+    this.cm2imap++;
+  },
+  
+  cm2imap:0,
+  
+  testConnexionImap: function(){
+    
+    if (Services.prefs.getBoolPref("courrielleur.testimap.startup")){
+      
+      cm2TestImapM2();
+    }
   },
 
   observe: function(aSubject, aTopic, aData)
@@ -123,11 +142,20 @@
     {
       Services.obs.addObserver(this, "profile-after-change");
       Services.obs.addObserver(this, "profile-change-net-teardown");
+      Services.obs.addObserver(this, "network:link-status-changed");
     }
     else if (aTopic == "profile-after-change")
     {
       this.onProfileStartup();
     }
+    else if ("network:link-status-changed"==aTopic)
+    {
+      Services.obs.removeObserver(this, "network:link-status-changed");
+
+      if (1==this.cm2imap)
+        this.testConnexionImap();
+      this.cm2imap++;
+    }
   },
 
 
